packages/webapp/src/containers/Animals/AddAnimals/useAnimalOptions.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import {
+  useGetDefaultAnimalTypesQuery,
+  useGetCustomAnimalTypesQuery,
+  useGetDefaultAnimalBreedsQuery,
+  useGetCustomAnimalBreedsQuery,
+  useGetAnimalSexesQuery,
+  useGetAnimalIdentifierTypesQuery,
+  useGetAnimalIdentifierColorsQuery,
+  useGetAnimalOriginsQuery,
+  useGetAnimalUsesQuery,
+} from '../../../store/api/apiSlice';
+import { useTranslation } from 'react-i18next';
+import { generateUniqueAnimalId } from '../../../util/animal';
+import { ANIMAL_ID_PREFIX } from '../types';
+
+type OptionType =
+  | 'default_types'
+  | 'type'
+  | 'breed'
+  | 'sex'
+  | 'sexDetails'
+  | 'use'
+  | 'tagType'
+  | 'tagColor'
+  | 'organicStatus'
+  | 'origin';
+
+export const useAnimalOptions = (...optionTypes: OptionType[]) => {
+  const { t } = useTranslation(['animal', 'common', 'translation']);
+
+  const { data: defaultTypes = [] } = useGetDefaultAnimalTypesQuery();
```

Comment:
As a rule, I don't use that hook because once you pass the queries through it, you lose the typing on the returned arrays. You could re-import the types manually but it kind of negates both the lines saved and the coolest part of RTK Query, which how the hooks kept track of the types defined in apliSlice (and there is always the possibility of introducing a bug when you're synching the types manually!)

The hook was also written in js before we shifted the frontend to (and wrote the apiSlice in) ts so we may just want to get rid of it.